test/languages/red_test.rb

```diff
@@ -0,0 +1,43 @@
+require "test_helper"
+
+module SnippetExtractor
+  module Languages
+    class RedTest < Minitest::Test
+      def test_full_example
+        code = <<~CODE
+          Red [] ; a comment
+
+          ; single line comment
+          x: 1 ; line comment 1
+          x: 2   ;-- line comment 2
+          x: 3		;@@ line comment 3
```

Comment:
```suggestion
          x: 1 ; line comment 1
          x: 2 ;-- line comment 2
          x: 3 ;@@ line comment 3
```

Reply:
it's intentional, but perhaps hard to deal with for the extractor. should it be seen as an extractor bug instead?

Reply:
First line should be left anyway, but it's not. It's not "off".

Reply:
Oh, of course it is, inside a herdoc string.  Sorry about that.  I did not look closely enough to get the context.